### PR TITLE
Refacto OpfabAPIService

### PR DESCRIPTION
**Pull request by freddidierRTE**
Move OpfabAPIService to a dedicate repository and name it OpfabAPI 
Split OpfabAPI for a better separation of concerns 
Remove unneeded declarations of api in opfab.js
Simplify some test initializations by using initOpfabAPIService()

Nothing in release note

### CODE REVIEW

**Review by vlo-rte — DISMISSED** [review]

File: ui/main/src/app/api/handlebars.api.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+import {LogOption, LoggerService as logger} from '../business/services/logs/logger.service';
+
+declare const opfab: any;
+
+export class HandlebarsAPI {
+    public static async getHandlebarHelpers() {
+        return undefined;
+    }
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/api/handlebars.api.ts:17]: it is important to have undefined by default

File: ui/main/src/app/api/opfab.api.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        opfab.alertMessage = {
+            messageLevel: Object.freeze(MessageLevel),
+            show(message, messageLevel) {
+                const msg = new Message(message, messageLevel);
+                return AlertMessageService.sendAlertMessage(msg);
+            }
+        };
+        Object.freeze(opfab.alertMessage);
+    }
+}
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/api/opfab.api.ts:54]: ok , done

File: ui/main/src/app/api/currentcard.api.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+ */
+
+/* Copyright (c) 2023-2024, RTE (http://www.rte-france.com)
+ * See AUTHORS.txt
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of the OperatorFabric project.
+ */
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/api/currentcard.api.ts:17]: outdated

File: ui/main/src/app/business/application-loader.ts [hunk 4]
```diff
@@ -301,12 +300,8 @@ export class ApplicationLoader {
...
 
     private initOpfabAPI(): void {
-        OpfabAPIService.init();
-        opfab.navigate.showCardInFeed = function (cardId: string) {
-            RouterService.navigateTo('feed/cards/' + cardId);
-        };
-
-        OpfabAPIService.initAPI();
+        OpfabAPI.init();
+        OpfabAPI.initAPI(this.translationService);
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/business/application-loader.ts:304]: not needed

File: ui/main/src/app/api/handlebars.api.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+    public static init() {
+        opfab.handlebars = {
+            registerCustomHelpers: function (getHandlebarHelpersFunction) {
+                HandlebarsAPI.getHandlebarHelpers = getHandlebarHelpersFunction;
+                logger.info('Registered function to get handlebar helpers', LogOption.LOCAL_AND_REMOTE);
+            }
+        };
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/api/handlebars.api.ts]: ok done

File: ui/main/src/app/api/businessconfig.api.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+            },
+
+            registerFunctionToGetTags: function (getTagsFunction) {
+                BusinessConfigAPI.getTags = getTagsFunction;
+                logger.info('Registered function to get tags', LogOption.LOCAL_AND_REMOTE);
+            }
+        };
+        // prevent unwanted modifications from templates code or custom scripts
+        Object.freeze(opfab.businessconfig);
+    }
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/api/businessconfig.api.ts]: not needed

File: ui/main/src/app/business/builtInTemplates/task/usercard/taskUserCardTemplateView.spec.ts [hunk 7]
```diff
@@ -7,31 +7,29 @@
...
 
 describe('Task UserCard Template View', () => {
     beforeEach(() => {
-        const translationService = new TranslationServiceMock();
-        OpfabAPIService.setTranslationService(translationService);
-        OpfabAPIService.init();
-        OpfabAPIService.initAPI();
-        OpfabAPIService.initUserCardTemplateInterface();
+        initOpfabAPIService();
+        CurrentUserCardAPI.initUserCardTemplateInterface();
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/business/builtInTemplates/task/usercard/taskUserCardTemplateView.spec.ts]: ok done

File: ui/main/src/app/modules/card/components/card-body/card-body.component.ts [hunk 8]
```diff
@@ -369,7 +369,7 @@ export class CardBodyComponent implements OnChanges, OnInit, OnDestroy {
         if (this.card.lttd && !this.lttdExpiredIsTrue && this.regularlyLttdCheckActive) {
             if (this.isLttdExpired()) {
                 this.lttdExpiredIsTrue = true;
-                OpfabAPIService.templateInterface.setLttdExpired(true);
+                CurrentCardAPI.templateInterface.setLttdExpired(true);
             } else setTimeout(() => this.regularlyCheckLttd(), 500);
         }
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/modules/card/components/card-body/card-body.component.ts:374]: you are right , there is a memory leak here
To be solved in another issue

File: ui/main/src/app/business/builtInTemplates/task/usercard/taskUserCardTemplateView.spec.ts [hunk 9]
```diff
@@ -7,31 +7,29 @@
...
 
 describe('Task UserCard Template View', () => {
     beforeEach(() => {
-        const translationService = new TranslationServiceMock();
-        OpfabAPIService.setTranslationService(translationService);
-        OpfabAPIService.init();
-        OpfabAPIService.initAPI();
-        OpfabAPIService.initUserCardTemplateInterface();
+        initOpfabAPI();
+        CurrentUserCardAPI.initUserCardTemplateInterface();
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/business/builtInTemplates/task/usercard/taskUserCardTemplateView.spec.ts]: done

File: ui/main/src/app/api/currentusercard.api.ts [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+                    CurrentUserCardAPI.currentUserCard.startDate = startDate;
+            },
+
+            setInitialSeverity: function (initialSeverity) {
+                CurrentUserCardAPI.currentUserCard.initialSeverity = initialSeverity;
+            },
+
+            setInitialKeepChildCards: function (initialKeepChildCards) {
+                CurrentUserCardAPI.currentUserCard.initialKeepChildCards = initialKeepChildCards;
+            },
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/api/currentusercard.api.ts:142]: has been put in another issue

File: ui/main/src/app/api/currentusercard.api.ts [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+            startDate: null,
+            userEntityChildCard: null,
+            selectedEntityRecipients: [],
+            selectedEntityForInformationRecipients: [],
+            setInitialSelectedRecipients: function (recipients) {},
+            setInitialSelectedRecipientsForInformation: function (recipients) {},
+            setSelectedRecipients: function (recipients) {},
+            setSelectedRecipientsForInformation: function (recipients) {},
+            setDropdownEntityRecipientList: function (recipients) {},
+            setDropdownEntityRecipientForInformationList: function (recipients) {}
```
[the remark this answers is not in the document]
Comment (freddidierRTE) [reply ui/main/src/app/api/currentusercard.api.ts:41]: not relevant

**Review by vlo-rte — APPROVED** [review]

**Pull request merged**